test/hack/resource/pkg/resourcetypes/eni.go

```diff
@@ -154,3 +160,59 @@ func (e *ENI) Cleanup(ctx context.Context, ids []string) ([]string, error) {
 
 	return deleted, errs
 }
+
+// get-vpc-resource-controller-enis is to get leaked ENIs by the vpc-resource-controller
+// Issue: https://github.com/aws/karpenter-provider-aws/issues/5582
+func (e *ENI) getVpcResourceControllerENIs(ctx context.Context, expirationTime time.Time, excludedClusters []string) (ids []string, err error) {
+	var nextToken *string
+	for {
+		out, err := e.cloudFormationClient.DescribeStacks(ctx, &cloudformation.DescribeStacksInput{
+			NextToken: nextToken,
+		})
+		if err != nil {
+			fmt.Println("dfdsf", err)
+			return ids, err
+		}
+
+		stacks := lo.Reject(out.Stacks, func(s cloudformationtypes.Stack, _ int) bool {
+			return s.StackStatus == cloudformationtypes.StackStatusDeleteComplete ||
+				s.StackStatus == cloudformationtypes.StackStatusDeleteInProgress ||
+				lo.FromPtr(s.CreationTime).After(expirationTime)
+		})
+		for _, stack := range stacks {
+			clusterName, found := lo.Find(stack.Tags, func(tag cloudformationtypes.Tag) bool {
+				return *tag.Key == karpenterTestingTag
+			})
+			if found {
+				if slices.Contains(excludedClusters, lo.FromPtr(clusterName.Value)) {
+					continue
+				}
+				out, err := e.ec2Client.DescribeNetworkInterfaces(ctx, &ec2.DescribeNetworkInterfacesInput{
+					Filters: []ec2types.Filter{
+						{
+							Name:   lo.ToPtr("tag-key"),
+							Values: []string{fmt.Sprintf("kubernetes.io/cluster/%s", lo.FromPtr(clusterName.Value))},
+						},
+						{
+							Name:   lo.ToPtr("tag:eks:eni:owner"),
```

Comment:
Which testing did you see NICs with this tag get created under? AFAIK, NICs with this tag are only created when using security groups for pods and branch ENIs (which I don't think we do in our testing). I thought the tags that the VPC CNI added for its NICs were: `node.k8s.amazonaws.com/instance_id` and `node.k8s.amazonaws.com/createdAt`